examples/esp_netif/slip_custom_netif/components/slip_modem/library/slip_modem.c

```diff
@@ -0,0 +1,246 @@
+/*
+ * SPDX-FileCopyrightText: 2020-2022 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <string.h>
+#include "slip_modem.h"
+
+#include "esp_netif.h"
+#include "slip_modem_netif.h"
+#include "esp_event.h"
+#include "esp_log.h"
+
+#define SLIP_RX_TASK_PRIORITY   10
+#define SLIP_RX_TASK_STACK_SIZE (4 * 1024)
+
+static const char *TAG = "slip-modem";
+
+
+// UART container object
+typedef struct {
+    // UART device number for SIO use
+    uart_port_t uart_dev;
+
+    // UART baud rate for configuration
+    uint32_t uart_baud;
+
+    // UART TX pin for configuration
+    int uart_tx_pin;
+
+    // UART RX pin for configuration
+    int uart_rx_pin;
+
+    // QueueHandle for uart driver
+    QueueHandle_t uart_queue;
+
+    // TaskHandle for receive task
+    TaskHandle_t uart_rx_task;
+} esp_slip_uart_t;
+
+
+// Modem object, implements glue logic for slip_driver and esp_netif
+struct slip_modem {
+    // ESP base netif driver
+    esp_netif_driver_base_t base;
+
+    // Uart for use with slip
+    esp_slip_uart_t uart;
+
+    // Buffer for incoming messages
+    uint8_t *buffer;
+    uint32_t buffer_len;
+
+    // Filter callbacks for application-specific slip message handling
+    slip_rx_filter_cb_t *rx_filter;
+
+    // Running flag
+    bool running;
+
+    // esp_netif related: SLIP interface IP6 address
+    esp_ip6_addr_t addr;
+
+};
+
+
+static void slip_modem_uart_rx_task(void *arg);
+static esp_err_t slip_modem_post_attach(esp_netif_t *esp_netif, void *args);
+
+// Create a new slip modem
+slip_modem_t *slip_modem_create(esp_netif_t *slip_netif, const slip_modem_config_t *modem_config)
+{
+    ESP_LOGI(TAG, "%s: Creating slip modem (netif: %p)", __func__, slip_netif);
+
+    ESP_LOGD(TAG, "%s (netif: %p)", __func__, slip_netif);
+
+    slip_modem_t *slip_modem = calloc(1, sizeof(slip_modem_t));
+    if (!slip_modem) {
+        ESP_LOGE(TAG, "create netif glue failed");
+        return NULL;
+    }
+
+    // Attach driver and post_attach callbacks
+    slip_modem->base.post_attach = slip_modem_post_attach;
+    slip_modem->base.netif = slip_netif;
+
+    // Attach config
+    slip_modem->buffer_len = modem_config->rx_buffer_len;
+
+    slip_modem->rx_filter = modem_config->rx_filter;
+
+    slip_modem->uart.uart_dev = modem_config->uart_dev;
+    slip_modem->uart.uart_baud = modem_config->uart_baud;
+    slip_modem->uart.uart_rx_pin = modem_config->uart_rx_pin;
+    slip_modem->uart.uart_tx_pin = modem_config->uart_tx_pin;
+    memcpy(&slip_modem->addr, modem_config->ipv6_addr, sizeof(esp_ip6_addr_t));
```

Comment:
Nitpick: This could be:
```cpp
slip_modem->addr = modem_config->ipv6_addr;
```

Reply:
I'm afraid this won't work, the config parameter is a pointer and the internal param is a struct.

Reply:
Of course, missed a `*`

```cpp
slip_modem->addr = *modem_config->ipv6_addr;
```

Reply:
Oh, can you actually assign `struct`s like that ? (and it would do the copy in the background?)